README.md

```diff
@@ -1 +1,291 @@
+
+In practice, that looks something like this:
+
+1. Canvas A supporting Feature X creates a canvas with a Feature X node in it and exports it as an Open Canvas file.
+2. Canvas B, which does not support Feature X, opens the Open Canvas file, and some edits are made to the canvas.
+3. Canvas B exports the canvas to an Open Canvas file. The nodes for Feature X should still be in the Open Canvas file, unchanged.
+
+NOTE: This is perhaps the most onerous part of supporting Open Canvas; implementing canvases have to store data that they don't need or recognize.
+
+### Extensibility through fallbacks
+
+The base node type supports an optional `fallback` property as a string. If a canvas wishes to visualize a node that it doesn't implement, the canvas can render a fallback element on the canvas at the element's x,y position using the string.
+
+### Extensibility through schema extension
+
+Schemas can extend one another.
+
+When extending, properties are simply merged to establish the full schema. This allows extension with meaningful fallbacks: if an implementer doesn't support the extended schema, but does support the base schema, they can render the node using a simpler representation.
+
+Schemas are defined in JSONSchema format.
+
+```json
+"schemas": [
+  "@ocwg/text": { /* JSONSchema format */ }
+]
+```
+
+For example, a `@tldraw/arrow` schema could extend a `@ocwg/arrow` schema, adding properties that are only applicable to TLDraw arrow.
+
+- [ ] Flesh out this example
+
+## Core Node Schemas
+
+These are the node schemas included in the core `@ocwg` namespace.
+
+### Text
+
+Text Nodes are used to display text on the canvas.
+
+### Assets
+
+Asset Nodes are used to represent files or other assets. They can be embedded or reference external sources. We use `mime-type` to describe the type of the asset and `uri` to describe the location of the asset.
+
+### Arrows
+
+### Examples
+
+Here are a bunch of example Nodes:
+
+```json
+{
+  "nodes": [
+    {
+      "id": "someShapeID11321",
+      "x": 10,
+      "y": 10,
+      "properties": [{
+        "schema": "@ocwg/text",
+        "schema_version": "1.0",
+        "text": "Hello, world!",
+        "color": "#000000"
+      }]
+    },
+    {
+      "id": "2435234634t324t3245",
+      "x": 10,
+      "y": 10,
+      "properties": [{
+        "schema": "@ocwg/text",
+        "schema_version": "1.0",
+        "w": 200,
+        "h": 150,
+        "mime_type": "text/json",
+        "uri": "data://9823fy9283hf2i3yf082ohu3fo2hufi2uh4fk234f8youhi",
+      }]
+    },
+    {
+      "id": "oajefioajef83r8y3ehf",
+      "x": 10,
+      "y": 10,
+      "properties": [{
+        "schema": "@ocwg/rectangle",
+        "schema_version": "1.0",
+        "width": 200,
+        "height": 150,
+        "text": "Hello, world!",
+        "stroke_color": "red",
+        "stroke_weight": ,
+        "fill_color": ,
+        "rotation": 90 /* +/-360 degrees */
```

Comment:
It's interesting to see `"rotation"` is under `"properties"`. Coming from glTF, this seems quite strange to me, since in glTF, all transform related properties (position, rotation, scale) are a part of the base node structure. For Open Canvas, I would expect to see position, rotation, scale, and size as part of the base node. If not all canvas items need those properties, there can be default values, such as a default rotation of `0.0`.

Reply:
That's a good point. We put them into `properties` because we didn't want canvases that don't support `rotation` to have to specify it. However, default values for those properties when not included addresses that concern for me.

I need to look at how json schema handles default values.

Reply:
Like this:

```json
"properties": {
    "rotation": {
        "type": "number",
        "description": "The rotation of this canvas node in radians.",
        "default": 0.0
    },
}
```

Reply:
A property that is required must be listed in the `"required"` array, like this:

```json
"required": [
    "type"
],
```

A non-required property should have a default value specified. Otherwise, it has an implicit default value, defaulting to "null" or similar. So non-required properties should usually have "default" but may not.

A property cannot both have a default value and be required, since that would be nonsense.